book_list.rb

```diff
@@ -0,0 +1,39 @@
+require 'json'
+require_relative './book'
+require_relative './label_list'
+
+class BookList
+  def initialize
+    file = File.open('books.json', 'a+')
+    @books = file.size.zero? ? [] : JSON.parse(file.read)
+    file.close
+  end
+
+  def show
+    file = File.open('books.json', 'r')
+    @books.each do |book|
+      puts "id: #{book['id']}\npublish date: #{book['publish_date']}\narchived: #{book['archived']}"
+      puts "publisher: #{book['publisher']}\ncover state: #{book['cover_state']}"
+      puts
+    end
+    file.close
+  end
+
+  def add
+    file = File.open('books.json', 'w')
+    book = Book.new(user_input('ID:'), user_input('publish date:'), user_input('archived:'), user_input('publisher'),
+                    user_input('cover state'))
+    book = book.to_json
+    @books.push(book)
+    file.write(JSON[@books])
+    file.close
+    puts 'Book created successfully'
+  end
```

Comment:
- [ ] Great job on implementing all the methods, your application works great 👍 According to the requirements however you should implement `can_be_archived?` which you created and is working however, you haven't implemented it in the `add` method yet, therefore whenever I create a book in your application, it always keeps the value that I entered for `archived`, even if the book is more than 10 years old or if the cover is bad.